src/lib/systemd.ts

```diff
@@ -1,113 +1,130 @@
 import * as Bluebird from 'bluebird';
-import * as dbus from 'dbus-native';
+import * as dbus from 'dbus';
+import TypedError = require('typed-error');
 
-const bus = dbus.systemBus();
-const invokeAsync = Bluebird.promisify(bus.invoke).bind(bus);
+import log from './supervisor-console';
 
-const defaultPathInfo = {
-	path: '/org/freedesktop/systemd1',
-	destination: 'org.freedesktop.systemd1',
-	interface: 'org.freedesktop.systemd1.Manager',
-};
+export class DbusError extends TypedError {}
 
-const systemdManagerMethodCall = (
-	method: string,
-	signature?: string,
-	body?: any[],
-	pathInfo: {
-		path: string;
-		destination: string;
-		interface: string;
-	} = defaultPathInfo,
-) => {
-	if (signature == null) {
-		signature = '';
+const bus = dbus.getBus('system');
+const getInterfaceAsync = Bluebird.promisify(bus.getInterface, {
+	context: bus,
+});
```

Comment:
NIT: you could use something like this to avoid repeating the try/catch below:

```typescript
async function getInterface(...args: Parameters<getInterfaceAsync>) {
    try {
        return await getInterfaceAsync(...args);
    } catch (e) {
        throw new DbusError(e);
    }
}
```